include/swift/SIL/SILConstants.h

```diff
@@ -0,0 +1,458 @@
+
+  /// Return just the memory object for an address value.
+  SymbolicValueMemoryObject *getAddressValueMemoryObject() const;
+
+  /// Produce an array of elements.
+
+  static SymbolicValue getArray(ArrayRef<SymbolicValue> elements,
+                                CanType elementType,
+                                llvm::BumpPtrAllocator &allocator);
+  static SymbolicValue
+  getArrayAddress(SymbolicValueMemoryObject *memoryObject) {
+    SymbolicValue result;
+    result.representationKind = RK_ArrayAddress;
+    result.value.directAddress = memoryObject;
+    return result;
+  }
+
+  ArrayRef<SymbolicValue> getArrayValue(CanType &elementType) const;
+
+  //===--------------------------------------------------------------------===//
+  // Helpers
+
+  /// Dig through single element aggregates, return the ultimate thing inside of
+  /// it.  This is useful when dealing with integers and floats, because they
+  /// are often wrapped in single-element struct wrappers.
+  SymbolicValue lookThroughSingleElementAggregates() const;
+
+  /// Given that this is an 'Unknown' value, emit diagnostic notes providing
+  /// context about what the problem is.  If there is no location for some
+  /// reason, we fall back to using the specified location.
+  void emitUnknownDiagnosticNotes(SILLocation fallbackLoc);
+
+  /// Clone this SymbolicValue into the specified allocator and return the new
+  /// version.  This only works for valid constants.
+  SymbolicValue cloneInto(llvm::BumpPtrAllocator &allocator) const;
+
+  void print(llvm::raw_ostream &os, unsigned indent = 0) const;
+  void dump() const;
+};
+
+static_assert(sizeof(SymbolicValue) == 2 * sizeof(void *),
+              "SymbolicValue should stay small and POD");
+
+inline llvm::raw_ostream &operator<<(llvm::raw_ostream &os, SymbolicValue val) {
+  val.print(os);
+  return os;
+}
+
+/// This is a representation of a memory object referred to be an address.
+/// Memory objects may be mutated over their lifetime, but their overall type
+/// remains the same.
+struct SymbolicValueMemoryObject {
+  Type getType() const { return type; }
+
+  SymbolicValue getValue() const { return value; }
+  void setValue(SymbolicValue newValue) { value = newValue; }
+
+  /// Create a new memory object whose overall type is as specified.
+  static SymbolicValueMemoryObject *create(Type type, SymbolicValue value,
+                                           llvm::BumpPtrAllocator &allocator);
+
+private:
+  const Type type;
+  SymbolicValue value;
+
+  SymbolicValueMemoryObject(Type type, SymbolicValue value)
+      : type(type), value(value) {}
+  SymbolicValueMemoryObject(const SymbolicValueMemoryObject &) = delete;
+  void operator=(const SymbolicValueMemoryObject &) = delete;
+};
+
+/// SWIFT_ENABLE_TENSORFLOW
+/// A graph operation attribute, used by GraphOperationInst.
+/// Attributes have a name and a constant value.
+struct GraphOperationAttribute {
+  Identifier name;
+  SymbolicValue value;
+};
+
```

Comment:
@marcrasi  Is this `GraphOperationAttribute` struct used anywhere in the interpreter? Can this be left out of the patch?

Reply:
It can be removed. I moved it out of this file on the tensorflow branch in preparation for this, but I must have accidentally copied this file over before I did the move.